neo/Consensus/ConsensusService.cs

```diff
@@ -375,8 +720,18 @@ private void OnTimer(Timer timer)
             }
             else if ((context.State.HasFlag(ConsensusState.Primary) && context.State.HasFlag(ConsensusState.RequestSent)) || context.State.HasFlag(ConsensusState.Backup))
             {
-                if (!context.State.HasFlag(ConsensusState.CommitSent))
+                if (context.State.HasFlag(ConsensusState.CommitSent))
```

Comment:
Nice catch. This answer part of the previous review about change the position of the CommitSent verification.

Reply:
We may want to have it periodically send recovery if this is happening also.

Reply:
Actually if we have it periodically send recovery it won't have to send `CommitSent` here.

Reply:
Also, if a node counts past the view where `CommitSent` was set by another node, having it periodically send recovery would likely recover it before the other node counted it's view again to trigger another recovery message to be sent to it. The only downside, is that it is a bigger payload, but I think the benefit outweighs the downside.

Reply:
I implemented the improvement I described here since it will improve the performance and help reduce the chance of a split scenario.